backend/src/main/java/gov/cdc/usds/simplereport/api/accountrequest/AccountRequestController.java

```diff
@@ -47,20 +82,119 @@ public void submitWaitlistRequest(@Valid @RequestBody WaitlistRequest body) thro
+        devices.stream()
+            .collect(Collectors.toMap(d -> d.getName(), d -> d.getInternalId().toString()));
+    Map<String, String> deviceModelsToIds =
+        devices.stream()
+            .collect(Collectors.toMap(d -> d.getModel(), d -> d.getInternalId().toString()));
+
+    List<String> testingDevicesSubmitted =
+        new ArrayList<>(Arrays.asList(reqVars.get("testingDevices").split(", ")));
+    testingDevicesSubmitted.removeIf(d -> d.toLowerCase().startsWith("other"));
+    List<String> testingDeviceIds =
+        testingDevicesSubmitted.stream()
+            .map(
+                d -> {
+                  String deviceId =
+                      Optional.ofNullable(deviceNamesToIds.get(d)).orElse(deviceModelsToIds.get(d));
+                  if (deviceId == null) {
+                    throw new RuntimeException(
+                        String.format("Submitted device=%s not registered in DB.", d));
+                  }
+                  return deviceId;
+                })
+            .collect(Collectors.toList());
+    String defaultTestingDeviceId =
+        Optional.ofNullable(deviceNamesToIds.get(reqVars.get("defaultTestingDevice")))
+            .orElse(deviceModelsToIds.get(reqVars.get("defaultTestingDevice")));
+    if (defaultTestingDeviceId == null) {
+      throw new RuntimeException(
+          String.format(
+              "Submitted default device=%s not registered in DB.",
+              reqVars.get("defaultTestingDevice")));
+    }
+    DeviceSpecimenTypeHolder deviceSpecimenTypes =
+        _dts.getTypesForFacility(defaultTestingDeviceId, testingDeviceIds);
+
+    StreetAddress facilityAddress =
+        _avs.getValidatedAddress(
+            reqVars.get("streetAddress1"),
+            reqVars.get("streetAddress2"),
+            reqVars.get("city"),
+            reqVars.get("state"),
+            reqVars.get("zip"),
+            _avs.FACILITY_DISPLAY_NAME);
+    StreetAddress providerAddress =
+        new StreetAddress(
+            Translators.parseString(reqVars.get("opStreetAddress1")),
+            Translators.parseString(reqVars.get("opStreetAddress2")),
+            Translators.parseString(reqVars.get("opCity")),
+            Translators.parseState(reqVars.get("opState")),
+            Translators.parseString(reqVars.get("opZip")),
+            Translators.parseString(reqVars.get("opCounty")));
+
+    PersonName providerName =
+        Translators.consolidateNameArguments(
+            null, reqVars.get("opFirstName"), null, reqVars.get("opLastName"), null, true);
+    PersonName adminName =
+        Translators.consolidateNameArguments(
+            null, reqVars.get("firstName"), null, reqVars.get("lastName"), null);
+
+    String orgExternalId =
+        String.format(
+            "%s-%s-%s",
+            reqVars.get("state"),
+            reqVars.get("organizationName").replace(' ', '-').replace(':', '-'),
+            UUID.randomUUID().toString());
```

Comment:
i think the thinking is that different organizations *can* in fact have the same name/externalID and the nonce serves to allow them to both exist. what behavior are you hoping for?

Reply:
If someone hits the submit button twice by mistake, I think the only uniqueness constraint they would hit is for user creation (where the email address must be unique).

It would be a slightly better experience if a double submission failed at this step, as that would result in duplicate emails sent and an orphaned organization resource in Okta. I'm not sure if there's anything in the organization resource that should be unique (maybe state + name + CLIA number?), so that's maybe something to ask product about and leave for now as a future enhancement.